libraries/eosiolib/contracts/eosio/multi_index.hpp

```diff
@@ -732,11 +732,21 @@ class multi_index
                _multidx->modify( *itr, payer, std::forward<Lambda&&>(updater) );
             }
 
+            template<typename Lambda>
+            void update( const_iterator itr, eosio::name payer, Lambda&& updater ) {
+               modify( itr, payer, std::forward<Lambda>(updater) );
+            }
+
             template<typename Lambda>
             void modify( const T& obj, eosio::name payer, Lambda&& updater ) {
                _multidx->modify( obj, payer, std::forward<Lambda&&>(updater) );
             }
 
+            template<typename Lambda>
+            void update( const T& obj, eosio::name payer, Lambda&& updater ) {
```

Comment:
`modify` has a small design flaw where it accepts `const` object and later modifies it using `const_cast`.
to not break existing contracts code we may leave `modify` as it is but for new interface I suggest we remove const-ness, it just confuses.